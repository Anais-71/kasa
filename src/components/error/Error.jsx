//CSS
import "./error.css"

function Error() {
    return (
        <div className="error">
            <h1 className="error__title">404</h1>
            <h2 className="error__subtitle">Oups! La page que vous demandez n'existe pas.</h2>
            <a className="error__link" href="my-app/public/index.html">Retourner sur la page d'accueil</a>
        </div>
    )
}

export default Error;