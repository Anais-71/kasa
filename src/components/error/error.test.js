import { render, screen } from '@testing-library/react'
import Error from './Error'

describe('Given my URL path is wrong', () => {
  beforeEach(() => {
    render(<Error />)
  })

  test('When the error component is being displayed, Then the 404 title should be present', () => {
    const titleElement = screen.getByText(/404/i)
    expect(titleElement).toBeInTheDocument()
  })

  test('When the error component is being displayed, Then the error message should be present', () => {
    const messageElement = screen.getByText(
      /Oups! La page que vous demandez n'existe pas./i,
    )
    expect(messageElement).toBeInTheDocument()
  })

  test('When the error component is being displayed, Then the home page link should be present and correct', () => {
    const linkElement = screen.getByText(/Retourner sur la page d'accueil/i)
    expect(linkElement).toBeInTheDocument()
    expect(linkElement).toHaveAttribute('href', '/')
  })

  test('The error page matches the snapshot', () => {
    const { asFragment } = render(<Error />)
    expect(asFragment()).toMatchSnapshot()
  })
})
