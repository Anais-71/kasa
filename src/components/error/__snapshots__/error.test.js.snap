// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given my URL path is wrong The error page matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="error"
  >
    <h1
      class="error__title"
    >
      404
    </h1>
    <h2
      class="error__subtitle"
    >
      Oups! La page que vous demandez n'existe pas.
    </h2>
    <a
      class="error__link"
      href="/"
    >
      Retourner sur la page d'accueil
    </a>
  </div>
</DocumentFragment>
`;
